lib/temporal/data_converter.rb

```diff
@@ -11,6 +11,10 @@ def initialize(payload_converter:, payload_codecs:, failure_converter:)
       @failure_converter = failure_converter
     end
 
+    def to_payload(data)
+      payload_converter.to_payload(data)
+    end
```

Comment:
This is confusing to me.
Why would the `DateConverter.to_payload` implementation ignore codecs?

Reply:
It just looks like a bug waiting to happen.
Nobody should use this method and arguably it shouldn't exist

Reply:
Ah, good point, it wasn't exposed publicly before. I'll add codecs 👍